src/Map/src/MapFactory.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\UX\Map;
+
+/**
+ * @author Simon André <smn.andre@gmail.com>
+ *
+ * @internal
+ */
+final class MapFactory
+{
+    /**
+     * @param array{
+     *  center?: array{lat: float, lng: float},
+     *  zoom?: float,
+     *  markers?: list<array{
+     *    position: array{lat: float, lng: float},
+     *    title?: string,
+     *    infoWindow?: array{
+     *      content?: string,
+     *      headerContent?: string,
+     *    },
+     *  }>,
+     * } $array The array representation of the map
+     */
+    public static function fromArray(array $array): Map
+    {
+        $map = new Map();
+
+        $map->fitBoundsToMarkers();
+
+        if (isset($array['center'])) {
+            if (!\is_array($array['center'])) {
+                throw new \InvalidArgumentException('The "center" parameter must be an array.');
+            }
+
+            $map->center(self::createPoint($array['center']));
+            $map->fitBoundsToMarkers(false);
+            unset($array['center']);
+        }
+
+        if (isset($array['zoom'])) {
+            if (!is_numeric($array['zoom'])) {
+                throw new \InvalidArgumentException('The "zoom" parameter must be numeric.');
+            }
+
+            $map->zoom((float) $array['zoom']);
+            $map->fitBoundsToMarkers(false);
+            unset($array['zoom']);
+        }
+
+        if (isset($array['markers'])) {
+            if (!\is_array($array['markers'])) {
+                throw new \InvalidArgumentException('The "markers" parameter must be an array.');
+            }
+            foreach ($array['markers'] as $marker) {
+                if (!\is_array($marker)) {
+                    throw new \InvalidArgumentException('The "markers" parameter must be an array of arrays.');
+                }
+                $marker = self::createMarker($marker);
+                $map = $map->addMarker($marker);
+            }
+            unset($array['markers']);
+        }
+
+        if (\count($array) > 0) {
+            throw new \InvalidArgumentException(\sprintf('Unknown map parameters: "%s"', implode(', ', array_keys($array))));
+        }
+
+        return $map;
+    }
+
+    private static function createMarker(array $marker): Marker
+    {
+        if (!\array_key_exists('position', $marker)) {
+            throw new \InvalidArgumentException('The "position" parameter is required.');
+        }
+        if (!\is_array($marker['position'])) {
+            throw new \InvalidArgumentException(\sprintf('The "position" parameter must be an array, "%s" given.', get_debug_type($marker['position'])));
+        }
```

Comment:
`createMarker` can be a hot point depending of markers count, do we really want to do all those checks? 
Couldn't we assume the user will use correct markers definition, and if it's not the case, we leverage all of this to PHP types instead?

Same comments for `infoWindow`, `title`, `extra`, etc... IMO you are doing too much things here, can't we simplify this code a lot?

Reply:
The main idea here is to provide a way to use "non-typed php objects" directly from Twig (hard-coded twig maps of data, json-like objects, etc).. so the performance trade-off is irrelevent imho in _this_ situation.

I wanted to simplify the data structure but decided to map the PHP classes for now, but something like this would be even better to me (and would require less code/checks/etc)

```twig
{{ ux_map({
    center: [4.51, 16.22],
    markers: [
       {lat: 45, lng: 22, content: 'sdsdf', id: 'store-22'}
   ]
)} }}
```

Also we 100% could accept "4.51 16.22" strings as point descriptor.

This is why a Factory made sense to me, for now and for the future. So now the (very valid) questions you raises here will help us decide what we need "now" and how we want to implement this!

---

In a near future i believe we will be able to offer dedicated tag/sub-component to declare markers, that will solve a lot of things DX-wise (and performance wise, as we then will be able to instanciate on-demand the infowindows)